<!doctype html><html><head><title>HIHISIV - storytelling</title><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta http-equiv=x-ua-compatible content="ie=edge"><link rel=stylesheet href=/css/bootstrap.min.css><link rel=stylesheet href=/css/layouts/main.css><link rel=stylesheet href=/css/navigators/navbar.css><link rel=stylesheet href=/css/plyr.css><link rel=stylesheet href=/css/flag-icon.min.css><link rel=stylesheet href="https://fonts.googleapis.com/css2?family=Muli:wght@300;400;500;600"><link rel=stylesheet href=/fontawesome/css/all.min.css><meta property="og:title" content="HIHISIV - storytelling"><meta property="og:description" content="storyteling hihisiv"><meta property="og:type" content="article"><meta property="og:url" content="https://quelopes.github.io/posts/introduction/"><meta property="article:published_time" content="2020-06-08T00:00:00+00:00"><meta property="article:modified_time" content="2020-06-08T00:00:00+00:00"><meta name=description content="storyteling hihisiv"><link rel=stylesheet href=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/styles/atom-one-dark.min.css><link rel=stylesheet href=/css/layouts/single.css><link rel=stylesheet href=/css/navigators/sidebar.css><link rel=stylesheet href=/css/style.css></head><body data-spy=scroll data-target=#TableOfContents data-offset=80><div class="container-fluid bg-dimmed wrapper"><nav class="navbar navbar-expand-xl top-navbar final-navbar shadow"><div class=container><button class="navbar-toggler navbar-light" id=sidebar-toggler type=button onclick=toggleSidebar()>
<span class=navbar-toggler-icon></span></button>
<a class=navbar-brand href=/>Raquel L. Costa</a>
<button class="navbar-toggler navbar-light" id=toc-toggler type=button onclick=toggleTOC()>
<span class=navbar-toggler-icon></span></button><div class="collapse navbar-collapse lang-selector" id=top-nav-items><ul class="navbar-nav ml-auto"></ul></div></div></nav><section class=sidebar-section id=sidebar-section><div class=sidebar-holder><div class=sidebar id=sidebar><form class=mx-auto method=get action=/search><input type=text name=keyword placeholder=Search data-search id=search-box></form><div class=sidebar-tree><ul class=tree id=tree><li id=list-heading><a href=/posts data-filter=all>Posts</a></li><div class=subtree><li><a class=active href=/posts/introduction/ title=HIHISIV>HIHISIV</a></li></div></ul></div></div></div></section><section class=content-section id=content-section><div class=content><div class="container p-0 read-area"><div class="hero-area col-sm-12" id=hero-area style=background-image:url(https://quelopes.github.io/images/default-hero.jpg)></div><div class=page-content><div class="author-profile ml-auto align-self-lg-center"><img class=rounded-circle src=/images/author/origamiScience_huc5a9e35ea0eaa7fa01b583bdeca60264_175655_120x120_fit_q75_box.jpg alt="Author Image"><h5 class=author-name>Raquel L. Costa</h5><p>:date_full</p></div><div class=title><h1>HIHISIV - storytelling</h1></div><div class=post-content id=post-content><h1 id=storytelling-about-hihisiv>Storytelling about HIHISIV</h1><p>The idea about this project appeared during my Ph.D. &ndash; when I collected and worked with transcriptome data from microarray platforms.
This page is more storytelling &ndash; I will describe the processes that I conducted and organized to generate the database HIHISIV.
In the end, I intend to point the difficulties and challenges faced as well as the importance of collaboration.</p><p>For academic information and technical description, please, check the preprint <a href=link>BioRxiv</a> and the code on <a href>Github</a>.</p><h2 id=data-organization>Data organization</h2><p>The first step of this story is the data survey in the main transcriptome repositories &ndash; GEO. Using the key-words <code>HIV</code> and <code>SIV</code> we found XX. The result was wrote in the Google sheet as show in the Figure X and posteriorly refined.</p><p><img src=/assets/images/googleSheet_hihisiv.png alt="first steps in the shared sheet."></p><p>\clearpage</p><p>The datasets were downloaded for a directory (CEL files).
In each project, a sheet was created (in .csv file) phenodata.</p><p>All material was organized as described in this structure below:</p><pre><code>hihisiv/

/hihisiv/      # hihisiv on the git
|--- a_code/
|    |--- microarray_analysis/
|    |--- rna-seq_analysis/
|         |--- get_sra/
|    |--- tables_organizing/
|--- b_database/
|    |--- tables_hihisiv/
          |--- gene_gene/
          |--- gene_species/
          |--- platform_transcript_id/
          |--- saved_google_sheet/
          |--- traits/
|    |--- Dockerfile
|    |--- init.sql
|--- c_web_app/
|    |--- hihisiv_jekyll/
|    |--- hihisiv_webapp/
|--- README.md             # description of the project
|--- docker-compose.yml
|--- metadata.html


/hihisiv_other_parts/
|--- d_documents/
|    |--- articles/
|    |--- manuscript/
|--- e_figures/
|--- f_datasets/
|    |--- microarray/
|    |--- rna-seq/
|--- g_data_results/
|    |--- microarray_results
|    |--- rna-seq_results
|--- storytelling.md

</code></pre><p>\newpage</p><h3 id=r-scripts>R scripts</h3><p><em>Microarray_analysis</em></p><table><thead><tr><th>script_name</th><th>input</th><th>output</th><th>library</th></tr></thead><tbody><tr><td>parameters.R</td><td>information about the experiment</td><td>-</td><td></td></tr><tr><td>module_processing.R</td><td>-</td><td>-</td><td></td></tr><tr><td>dependencies.R</td><td></td><td></td><td></td></tr><tr><td>raw_activity.R</td><td>raw data (CEL files)</td><td>normalized matrix</td><td>affy, impute</td></tr><tr><td>e-set_activity.R</td><td>normalized matrix</td><td>eSet object</td><td>-</td></tr><tr><td>limma_activity.R</td><td>eSet object and phenodata matrix</td><td>differentially expressed genes matrix</td><td>limma, ggplot2, RColorBrewer</td></tr></tbody></table><p><em>rna-seq_analysis</em></p><table><thead><tr><th>script_name</th><th>input</th><th>output</th><th>library</th></tr></thead><tbody><tr><td>get_sra.sh</td><td>archive with ids</td><td>SRA data</td><td>aria2c from aria2</td></tr><tr><td>sra_to_fastq.sh</td><td>SRA data</td><td>fastq files</td><td>fastq-dump from sratoolkit (v. 2.11.3)</td></tr><tr><td>quality_control.sh</td><td>fastq</td><td>.html reports</td><td>fastqc multiqc</td></tr><tr><td>alignment_rsem.sh</td><td>fastq paired</td><td></td><td>bowtie2; perl; rsem</td></tr><tr><td>limma_voom.R</td><td></td><td></td><td></td></tr><tr><td>rsem_matrix.sbatch</td><td>*genes.results</td><td>geneMat.txt</td><td>bowtie2; rsem</td></tr></tbody></table><table><thead><tr><th>tables and list</th><th>input</th></tr></thead><tbody><tr><td>GSE119234_hihisiv.txt</td><td>list of the samples with url. Used by script.</td></tr><tr><td>GSE119234_metadata.csv</td><td>information about samples (phenodata).</td></tr></tbody></table><h2 id=docker-containers>Docker containers</h2><h3 id=database-docker>Database docker</h3><ul><li>Build database docker container (remove old containers and images if needed):</li></ul><p><code>cd B_Database/</code>
<code>docker build -t hihisiv-postgres .</code></p><ul><li>Create empty directory on host to store database files, for example:</li></ul><p><code>mkdir /home/user/hihisiv_db</code></p><h3 id=web-app>Web-app</h3><ul><li>Build web application docker container (remove old containers and images if needed):</li></ul><p><code>cd c_web_app/hihisiv_webapp</code></p><p><code>docker build -t hihisiv-webapp .</code></p><h3 id=web-static>Web-static</h3><p><code>bundle exec jekyll serve</code></p><pre><code>/home/
|--- Background # about HIV and SIV 
|--- Experiments list # normalized dataset to download
|--- Database # postgres, streamlit, database in fact
|--- Documentation
|--- Data availability ????
|--- External sources

</code></pre><h3 id=all-in-one-docker-compose>All-in-one, docker-compose</h3><p>Requirements: docker and docker-composed</p><p>Installation steps:</p><ul><li>Edit web application container environment variables to enter database initialization file directory (<code>b_database/tables_hihisiv</code>) and database files directory:</li></ul><p><code>vi .env</code></p><ul><li><p>Set <code>VIRTUAL_HOST</code> on <code>docker-compose.yml</code> to host name of the host that will run the web application.</p></li><li><p>Start containers with:</p></li></ul><p><code>docker-compose up -d</code></p><ul><li>Open on the browser:</li></ul><p><code>localhost</code></p><ul><li>Stop docker:</li></ul><p><code>docker-compose down</code></p><h2 id=documents-manuscript>Documents, manuscript</h2><p>Contem artigos</p><ul><li>Manuscrito no Overleaf</li></ul><h2 id=figures>Figures</h2><p>Figuras geradas no google slides:</p><ul><li>logo HIHISIV</li><li>dataflow analysis</li><li>framework</li></ul><p>Modelo conceitual gerado no XXX</p><h2 id=frustations-difficults-challenges-and-conclusions>Frustations, Difficults, challenges and conclusions</h2><p>HIHISIV old version</p><p><img src=e_figures/old_conceptual_model_hihisiv.png alt=Graph.></p><p><img src=e_figures/old_mainPage_hihisiv.png alt=Graph.>
<img src=e_figures/old_framework_hihisiv.jpg alt=Graph.></p><h2 id=commands>Commands</h2><p><em>Postgres - main commands</em></p><pre><code>* psql hihisiv
* \d schema
* \d virus insert
* psql -f arq.sql hihisiv
* copy experiment form &quot;arq.csv&quot; delimiter ',' csv
* select * from design;
* delete from XX; todo o conteúdo da tabela é deletado.
* dropdb hihisiv
* createdb hihisiv
* barra q quit the database
* drop table XX;
* drop table XXX cascade;  
</code></pre><p><em>Comandos em docker</em></p><pre><code>* docker images
* docker ps
* docker stop &lt;container_id&gt;  
* docker rmi --force &lt;container_id&gt;
</code></pre></div><div class="row pl-3 pr-3"><div class="col-md-6 share-buttons"></div><div class="col-md-6 btn-improve-page"><a href=https://github.com/quelopes/quelopes.github.io/edit/main/content/posts/introduction/index.md title="Improve this page" target=_blank rel=noopener><i class="fas fa-code-branch"></i>Improve this page</a></div></div><hr><div class="row next-prev-navigator"></div><hr></div></div></div><a id=scroll-to-top class=btn><i class="fas fa-chevron-circle-up"></i></a></section><section class=toc-section id=toc-section><div class=toc-holder><h5 class="text-center pl-3">Table of Contents</h5><hr><div class=toc><nav id=TableOfContents><ul><li><a href=#data-organization>Data organization</a><ul><li><a href=#r-scripts>R scripts</a></li></ul></li><li><a href=#docker-containers>Docker containers</a><ul><li><a href=#database-docker>Database docker</a></li><li><a href=#web-app>Web-app</a></li><li><a href=#web-static>Web-static</a></li><li><a href=#all-in-one-docker-compose>All-in-one, docker-compose</a></li></ul></li><li><a href=#documents-manuscript>Documents, manuscript</a></li><li><a href=#figures>Figures</a></li><li><a href=#frustations-difficults-challenges-and-conclusions>Frustations, Difficults, challenges and conclusions</a></li><li><a href=#commands>Commands</a></li></ul></nav></div></div></section></div><footer id=footer class="container-fluid text-center align-content-center footer pb-2"><div class="container pt-5"><div class="row text-left"><div class="col-md-4 col-sm-12"><h5>Navigation</h5><ul><li class=nav-item><a class=smooth-scroll href=https://quelopes.github.io#about>About</a></li><li class=nav-item><a class=smooth-scroll href=https://quelopes.github.io#skills>Skills</a></li><li class=nav-item><a class=smooth-scroll href=https://quelopes.github.io#experiences>Experiences</a></li><li class=nav-item><a class=smooth-scroll href=https://quelopes.github.io#projects>Projects</a></li><li class=nav-item><a class=smooth-scroll href=https://quelopes.github.io#recent-posts>Projects</a></li></ul></div><div class="col-md-4 col-sm-12"><h5>Contact me:</h5><ul><li><a href=mailto:quelopes@gmail.com target=_blank rel=noopener><span><i class="fas fa-envelope"></i></span><span>quelopes@gmail.com</span></a></li></ul></div><div class="col-md-4 col-sm-12"><p>Stay up to date with email notification</p><form method=post action=https://blogtrottr.com><div class=form-group><input type=email class=form-control name=btr_email placeholder="Enter email"><br><input type=hidden name=btr_url value=https://quelopes.github.ioindex.xml>
<input type=hidden name=schedule_type value=1>
<small id=emailHelp class="form-text text-muted">By entering your email address, you agree to receive the newsletter of this website.</small>
<button type=submit class="btn btn-info"> Submit</button></div></form></div></div></div><hr><div class=container><div class="row text-left"><div class=col-md-4><a id=theme href=https://github.com/hossainemruz/toha target=_blank rel=noopener><img src=/images/theme-logo_hu8376fd15465fef26ffe66b6bcf0ca686_13669_32x0_resize_box_2.png alt="Toha Theme Logo">
Toha</a></div><div class="col-md-4 text-center">© 2020 Copyright.</div><div class="col-md-4 text-right"><a id=hugo href=https://gohugo.io/ target=_blank rel=noopener>Powered by
<img src=/images/hugo-logo.svg alt="Hugo Logo" height=18></a></div></div></div></footer><script type=text/javascript src=/js/jquery-3.4.1.min.js></script><script type=text/javascript src=/js/popper.min.js></script><script type=text/javascript src=/js/bootstrap.min.js></script><script type=text/javascript src=/js/navbar.js></script><script type=text/javascript src=/js/plyr.js></script><script type=text/javascript src=/js/main.js></script><script src=https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.18.1/highlight.min.js></script><script src=/js/single.js></script><script>hljs.initHighlightingOnLoad();</script></body></html>